backend/database_handler/llm_providers.py

```diff
@@ -0,0 +1,76 @@
+from backend.domain.types import LLMProvider
+from backend.node.create_nodes.providers import get_default_providers
+from .models import LLMProviderDBModel
+from sqlalchemy.orm import Session
+
+
+class LLMProviderRegistry:
+    def __init__(self, session: Session):
+        self.session = session
+
+    def reset_defaults(self):
+        """Reset all providers to their default values."""
+        self.session.query(LLMProviderDBModel).delete()
+
+        providers = get_default_providers()
+        for provider in providers:
+            self.session.add(_to_db_model(provider))
+
+        self.session.commit()
+
+    def get_all(self) -> list[LLMProvider]:
+        return [
+            _to_domain(provider)
+            for provider in self.session.query(LLMProviderDBModel).all()
+        ]
+
+    def get_all_dict(self) -> list[dict]:
+        return [
+            {
+                "id": provider.id,
+                "provider": provider.provider,
+                "model": provider.model,
+                "config": provider.config,
+            }
+            for provider in self.session.query(LLMProviderDBModel).all()
+        ]
```

Comment:
this is duplicated since `LLMProvider` does not have an ID. I think that it's fine for now, but I'm open to opinions on "should the domain models have IDs?"